File: sdk/python/examples/scrobble.py
"""Submit a scrobble.

Reads the bearer token from $ROCKSKY_TOKEN. Override the API base URL with
$ROCKSKY_BASE_URL if you're hitting a self-hosted instance.

Run:
    ROCKSKY_TOKEN=... uv run python examples/scrobble.py
"""

from __future__ import annotations

import asyncio
import os
import sys
import time

from rocksky import APIError, Client


async def main() -> int:
    token = os.environ.get("ROCKSKY_TOKEN")
    if not token:
        print("set $ROCKSKY_TOKEN first")
        return 1

    base_url = os.environ.get("ROCKSKY_BASE_URL", "https://api.rocksky.app")

    async with Client(base_url=base_url, token=token) as rocksky:
        try:
            result = await rocksky.scrobble.create(
                title="Hounds of Love",
                artist="Kate Bush",
                album="Hounds of Love",
                duration=298_000,
                year=1985,
                timestamp=int(time.time()),
            )
        except APIError as exc:
            print(f"scrobble failed: {exc}")
            return 1

    print("scrobble accepted:", result)
    return 0


if __name__ == "__main__":
    sys.exit(asyncio.run(main()))

File: sdk/rust/examples/scrobble.rs
//! Submit a scrobble.
//!
//! Reads the bearer token from `$ROCKSKY_TOKEN`. Override the API base URL
//! with `$ROCKSKY_BASE_URL` for self-hosted instances.
//!
//! ```text
//! ROCKSKY_TOKEN=... cargo run --example scrobble
//! ```

use std::time::{SystemTime, UNIX_EPOCH};

use rocksky::Client;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let token = match std::env::var("ROCKSKY_TOKEN") {
        Ok(t) => t,
        Err(_) => {
            eprintln!("set $ROCKSKY_TOKEN first");
            std::process::exit(1);
        }
    };
    let base_url = std::env::var("ROCKSKY_BASE_URL").unwrap_or_else(|_| "https://api.rocksky.app".into());

    let client = Client::builder().base_url(base_url).token(token).build();

    let now = SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs() as i64;

    let result = client
        .scrobble()
        .create("Hounds of Love", "Kate Bush")
        .album("Hounds of Love")
        .duration(298_000)
        .year(1985)
        .timestamp(now)
        .send()
        .await?;

    println!("scrobble accepted: {result}");
    Ok(())
}
